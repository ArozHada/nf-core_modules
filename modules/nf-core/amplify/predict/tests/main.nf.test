nextflow_process {

    name "Test Process AMPLIFY_PREDICT"
    script "../main.nf"
    process "AMPLIFY_PREDICT"

    tag "modules"
    tag "modules_nfcore"
    tag "amplify"
    tag "amplify/predict"
    tag "pyrodigal"
    tag "gunzip"

    setup {
            run("PYRODIGAL") {
                script "../../../pyrodigal/main.nf"
                process {
                    """
                    input[0] =  Channel.fromList([
                        tuple([ id:'test', single_end:false ], // meta map
                        file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true))
                    ])
                    """
                }
            }
            run("GUNZIP") {
                script "../../../gunzip/main.nf"
                process {
                    """
                    input[0] =  PYRODIGAL.out.faa

                    """
                }
            }
        }

   test("sarscov2 - contigs [fasta]") {

        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - contigs [fasta] - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.tsv.collect { file(it[1]).getName() } +
                                process.out.versions).match() }
            )
        }

    }

}
