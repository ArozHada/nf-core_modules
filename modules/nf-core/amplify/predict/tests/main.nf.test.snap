{
    "sarscov2 - contigs [fasta] - stub": {
        "content": [
            [
                "test.tsv",
                "versions.yml:md5,5fccbfe34bc2894950abea0d409911b4"
            ]
        ],
        "timestamp": "2023-12-26T20:47:26.54257423"
    },
    "sarscov2 - contigs [fasta]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,1951084ce1d410028be86754997e5852"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5fccbfe34bc2894950abea0d409911b4"
                ],
                "tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,1951084ce1d410028be86754997e5852"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5fccbfe34bc2894950abea0d409911b4"
                ]
            }
        ],
        "timestamp": "2023-12-26T20:47:09.588509017"
    }
}